packages/core/src/storages/request_queue_v2.ts

```diff
@@ -154,13 +150,80 @@ export class RequestQueue extends RequestProvider {
         //    will not put the request again to queueHeadDict.
         if (request.handledAt) {
             this.log.debug('Request fetched from the beginning of queue was already handled', { nextRequestId });
-            this.recentlyHandledRequestsCache.add(nextRequestId, true);
             return null;
         }
 
         return request;
     }
 
+    /**
+     * @inheritDoc
+     */
+    override async isFinished(): Promise<boolean> {
```

Comment:
@drobnikj I didn't remove the inheritance from `RequestProvider` completely, just overwrote this method. I don't think there's any other stuff in `RequestProvider` that could cause any trouble, but feel free to prove me wrong.

Reply:
As I wrote, I would remove original implementation from Provider do not to confuse future developers.